app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt

```diff
@@ -0,0 +1,202 @@
+        assertThat(launcherPackage, notNullValue())
+        device.wait(Until.hasObject(By.pkg(launcherPackage).depth(0)),
+            TIMEOUT
+        )
+
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        val intent = context.packageManager.getLaunchIntentForPackage("com.example.campusguide")
+        context.startActivity(intent)
+
+        device.wait(Until.hasObject(By.pkg("com.example.campusguide").depth(0)),
+            TIMEOUT
+        )
+    }
+
+    @Test
+    fun getDirectionsToConcordiaFromCurrentLocation() {
+
+        // Wait until the map of MapsActivity is loaded
+        device.wait(Until.hasObject(By.desc("maps_activity_map ready")), TIMEOUT)
+
+        // Click the Navigate button
+        val navigateButton: UiObject = device.findObject(UiSelector().descriptionContains("navigateButton"))
+        if(navigateButton.exists() && navigateButton.isEnabled) {
+            navigateButton.click()
+        }
+
+        // Wait until the ChooseDestinationOptions dialog appears
+        device.wait(Until.hasObject(By.desc("chooseDestinationOptionsLayout")), TIMEOUT)
+
+        // Click on the Search For Location button
+        val searchForLocationButton: UiObject = device.findObject(UiSelector().descriptionContains("searchForLocationButton"))
+        if(searchForLocationButton.exists() && searchForLocationButton.isEnabled) {
+            searchForLocationButton.click()
+        }
+
+        // Wait until the CustomSearchActivity activity is launched
+        device.wait(Until.hasObject(By.desc("customSearchLayout")), TIMEOUT)
+
+        // Search for Concordia University
+        // Note: For some reason, the regular setText() method doesn't work on a SearchView
+        // So I had to resort to legacySetText()
+        val searchTextField: UiObject = device.findObject(UiSelector().descriptionContains("searchView"))
+        searchTextField.legacySetText("Concordia University")
+
+        // Select the first option
+        val firstResult: UiObject = device.findObject(UiSelector().className("android.widget.TextView").text("Concordia University"))
+        if (firstResult.exists() && firstResult.isEnabled) {
+            firstResult.click()
+        }
+
+        // Wait until the ChooseOriginOptions dialog appears
+        device.wait(Until.hasObject(By.desc("chooseOriginOptionsLayout")), TIMEOUT)
+
+        // Click on the Use Current Location button
+        val useCurrentLocationButton: UiObject = device.findObject(UiSelector().descriptionContains("useCurrentLocationButton"))
+        if(useCurrentLocationButton.exists() && useCurrentLocationButton.isEnabled) {
+            useCurrentLocationButton.click()
+        }
+
+        // Wait until the map of DirectionsActivity is loaded
+        device.wait(Until.hasObject(By.desc("directions_activity_map ready")), TIMEOUT)
+
+        // Pick a random travel mode
+        val travelMode = travelModes.random()
```

Comment:
There is a bit of debate surrounding randomness in tests. Personally, i'd rather we remove this kind of randomness and always use one particular travel mode.

The reasoning behind this is that if the feature is broken for, say, only "driving" mode, then the test will be flaky and fail 33% of the time. When it does fail intermittently, we'll have no idea why until we dig in and find this element of randomness, which could waste a lot of time. It could also lie unnoticed through a merge into Master, and the dev who notices it may not be the dev who broke it.

All this to say: just put travelMode = "walking" or something

Reply:
I agree with most of what you said
However, functionally all three travel mode buttons do the same thing  (they make an API request and display the route) so if one of them works, then most likely all of them will (and vice versa) which is why I decided to go with the randomness approach and avoid writing duplicated code to test every travel mode